smartsim/settings/base.py

```diff
@@ -448,12 +455,16 @@ def add_exe_args(self, args: t.Union[str, t.List[str]]) -> None:
         :type args: str | list[str]
         :raises TypeError: if exe args are not strings
         """
+
+        if not isinstance(args, (list, str)):
```

Comment:
Two comments here:

1.  This TypeError message is not really correct given this statement.  
2.  Given that the ``exe_args`` setter method is doing the same check as ``add_exe_args`` should we embed the type check in an internal class method like ``_check_exe_arg_type(args)`` or something similar?  Open to suggestions, but I think the checks in this routine need to be improved (at least group all of the checks at the top of the method and mimic the other method)